src/components/sets/sets.js

```diff
@@ -0,0 +1,374 @@
+/* eslint-disable no-prototype-builtins */
+import uuidv4 from 'uuid/v4';
+// Storage of hierarchical sets of IDs (cell IDs, gene IDs, etc...).
+
+const CURRENT_SET_NAME = 'Current Selection';
+const ALL_ROOT_KEY = 'all';
+const ALL_ROOT_NAME = 'All';
+
+export class HSetsNode {
+  constructor(props) {
+    const {
+      setKey,
+      name,
+      editing = false,
+      isCurrentSet = false,
+      wasPreviousCurrentSet = false,
+      color,
+      children,
+      set,
+    } = props || {};
+    this.setKey = setKey;
+    this.name = name;
+    this.set = set;
+    this.children = children;
+    this.color = color;
+    this.editing = editing;
+    this.isCurrentSet = isCurrentSet;
+    this.wasPreviousCurrentSet = wasPreviousCurrentSet;
+  }
+
+  startEditing() {
+    this.editing = true;
+  }
+
+  stopEditing() {
+    this.editing = false;
+  }
+
+  isEditing() {
+    return this.editing;
+  }
+
+  setChildren(children) {
+    this.children = children;
+  }
+
+  setName(name) {
+    this.name = name;
+  }
+
+  setIsCurrentSet(v) {
+    this.isCurrentSet = v;
+  }
+
+  setWasPreviousCurrentSet(v) {
+    this.wasPreviousCurrentSet = v;
+  }
+
+  isLeaf() {
+    return (this.children === undefined
+        || this.children === null
+        || (Array.isArray(this.children) && this.children.length === 0));
+  }
+
+  findNode(setKey) {
+    if (this.setKey === setKey) {
+      return this;
+    }
+    if (!this.children) {
+      return null;
+    }
+    // eslint-disable-next-line no-restricted-syntax
+    for (const child of this.children) {
+      const childResult = child.findNode(setKey);
+      if (childResult) {
+        return childResult;
+      }
+    }
+    return null;
+  }
+
+  findParentNode(setKey) {
+    if (!this.children || this.children.length === 0) {
+      return null;
+    }
+    // eslint-disable-next-line no-restricted-syntax
+    for (const child of this.children) {
+      if (child.setKey === setKey) {
+        return this;
+      }
+      const childResult = child.findParentNode(setKey);
+      if (childResult) {
+        return childResult;
+      }
+    }
+    return null;
+  }
+
+  findCurrentSetNode() {
+    if (this.isCurrentSet) {
+      return this;
+    }
+    if (!this.children) {
+      return null;
+    }
+    // eslint-disable-next-line no-restricted-syntax
```

Comment:
Or maybe see if just the part of `no-restricted-syntax` you don't need could be disabled globally, if you think that's safe?